File: AcupayEmployee/Core Forms/PayStub/Entity/ShiftSchedule.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("employeeshift")>
    Public Class ShiftSchedule

        <Key>
        Public Property RowID As Integer?

        Public Property Created As Date

        Public Property CreatedBy As Integer?

        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        Public Property EmployeeID As Integer?

        Public Property ShiftID As Integer?

        Public Property EffectiveFrom As Date

        Public Property EffectiveTo As Date

        <Column("NightShift")>
        Public Property IsNightShift As Boolean

        <Column("RestDay")>
        Public Property IsRestDay As Boolean

        <ForeignKey("ShiftID")>
        Public Overridable Property Shift As Shift

    End Class

End Namespace

File: Accupay.Data/Entities/ShiftSchedule.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Data.Entities
{
    [Table("employeeshift")]
    public class ShiftSchedule
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? RowID { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        public int OrganizationID { get; set; }

        public int? EmployeeID { get; set; }

        public int? ShiftID { get; set; }

        public DateTime EffectiveFrom { get; set; }

        public DateTime EffectiveTo { get; set; }

        [Column("NightShift")]
        public bool IsNightShift { get; set; }

        [Column("RestDay")]
        public bool IsRestDay { get; set; }

        public bool IsWorkingDay => !IsRestDay;

        [ForeignKey("ShiftID")]
        public virtual Shift Shift { get; set; }
    }
}